### DISCUSSION TITLE
Where to add vim.filetype options

### DISCUSSION THREAD

**Discussion by JSaterdalen** [LazyVim/LazyVim#2490] [Q&A]
I'd like to add the following configuration so my dotfiles without an extension are treated like bash files. 

I added the following to lua/config/options.lua:

```lua
vim.filetype.add({
  pattern = {
    [".*/zsh/.*"] = "bash",
  },
})
```

This doesn't seem to work. Is this the correct location?

**Comment by dpetka2001** [comment]
I don't believe it's a matter of location, rather a matter of the lua pattern matching. Unfortunately, my lua-fu is also lackluster and couldn't make it work for filenames without extension. I added doing it with an auto-command like this
```lua
vim.api.nvim_create_autocmd({ "BufNewFile", "BufRead" }, {
  group = vim.api.nvim_create_augroup("filetype_bash", { clear = true }),
  desc = "Change filetype to bash",
  callback = function()
    if vim.fn.expand("%:e") == "" then
      vim.bo.filetype = "bash"
    end
  end,
})
```
Feel free to make any changes you'd like. This auto-command will take effect for every file without extension, so if you'd only want specific files in a directory to take effect, you could add an extra check with something like
```lua
if vim.fn.getcwd() == vim.fn.stdpath("config") then  -- change here with a condition for the directory you'd like
	-- the previous callback snippet from the autocommand
end
```
